modules/nextflow/src/main/groovy/nextflow/script/ProcessEntryHandler.groovy

```diff
@@ -0,0 +1,210 @@
+     * @return List of input parameter names found in the process
+     */
+    private List<String> parseProcessInputNames(ProcessDef processDef) {
+        def inputNames = []
+        
+        // Create delegate to capture Nextflow's internal input method calls
+        def delegate = new Object() {
+            def _in_val(tokenVar) { inputNames.add(tokenVar.name.toString()) }
+            def _in_path(tokenVar) { inputNames.add(tokenVar.name.toString()) }
+            def _in_file(tokenVar) { inputNames.add(tokenVar.name.toString()) }
+            def _in_env(tokenVar) { inputNames.add(tokenVar.name.toString()) }
+            def _in_each(tokenVar) { inputNames.add(tokenVar.name.toString()) }
+            
+            def _in_tuple(Object... items) {
+                for( item in items ) {
+                    if( item?.hasProperty('name') ) {
+                        inputNames.add(item.name.toString())
+                    }
+                }
+            }
+            
+            // Handle legacy input block syntax for backward compatibility
+            def input(Closure inputBody) {
+                def inputDelegate = new Object() {
+                    def val(name) { inputNames.add(name.toString()) }
+                    def path(name) { inputNames.add(name.toString()) }
+                    def file(name) { inputNames.add(name.toString()) }
+                    def env(name) { inputNames.add(name.toString()) }
+                    def each(name) { inputNames.add(name.toString()) }
+                    def tuple(Object... items) {
+                        for( item in items ) {
+                            if( item instanceof String || (item instanceof groovy.lang.GString) ) {
+                                inputNames.add(item.toString())
+                            }
+                        }
+                    }
+                    def methodMissing(String name, args) {
+                        for( arg in args ) {
+                            if( arg instanceof String || (arg instanceof groovy.lang.GString) ) {
+                                inputNames.add(arg.toString())
+                            }
+                        }
+                    }
+                }
+                inputBody.delegate = inputDelegate
+                inputBody.resolveStrategy = Closure.DELEGATE_FIRST
+                inputBody.call()
+            }
+            
+            // Ignore all other method calls during parsing
+            def methodMissing(String name, args) { /* ignore */ }
+        }
+        
+        // Execute the process body with our capturing delegate
+        def bodyClone = processDef.rawBody.clone()
+        bodyClone.delegate = delegate
+        bodyClone.resolveStrategy = Closure.DELEGATE_FIRST
+        
+        try {
+            bodyClone.call()
+        } catch (Exception e) {
+            // Ignore exceptions during parsing - we only want to capture input names
+        }
```

Comment:
@bentsherman i'm curious about your opinion about this? I believe it should be possible to use the ProcessDef to retrieve this metadata. WDT?